WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -763,6 +760,9 @@ public void onTrimMemory(final int level) {
                     break;
                 case TRIM_MEMORY_BACKGROUND:
                 case TRIM_MEMORY_UI_HIDDEN:
+                    // See https://android.jlelse.eu/how-to-detect-android-application-open-and-close-background-and-foreground-events-1b4713784b57
+                    onAppGoesToBackground();
```

Comment:
If we are not sure that this is called only once before the application `onActivityResumed` we might want to check for `sAppIsInTheBackground` in `onAppGoesToBackground` and only handle do the rest if it was not already `true`. It'd act as sanity check and we wouldn't need to care too much about double callbacks.

Having said that, since @aforcier mentioned that they are using this approach in Woo already, this might not be necessary. I am good with it either way.

Edit: I originally had a question about removing `TRIM_MEMORY_BACKGROUND` as well, but it looks like it was removed before I could post it 😝

Reply:
I'm a fan of adding a safety net :)  Good call